webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -821,6 +966,87 @@ private SmartlingFile processTranslationBatch(
     }
   }
 
+  private SmartlingFile uploadAiTranslationFile(
+      List<TextUnitDTO> batch,
+      String localeTag,
+      String fileName,
+      AndroidStringDocumentMapper mapper,
+      Repository repository,
+      String projectId,
+      SmartlingOptions options,
+      Map<String, String> localeMapping,
+      Set<Long> filterTmTextUnitIds) {
+
+    try (var timer =
+        Timer.resource(meterRegistry, "SmartlingSync.processTranslationBatch")
+            .tag("repository", repository.getName())
+            .tag("locale", localeTag)) {
+
+      logger.debug("Process translation batch for file: {}", fileName);
+      List<TextUnitDTO> fileBatch = batch;
+      SmartlingFile file = new SmartlingFile();
+      file.setFileName(fileName + "_" + localeTag);
```

Comment:
Do we rely on this specific filename construction anywhere else? (i.e.: needing to reconstruct it anywhere downstream?)

Reply:
This is only needed for the SmartlingResultProcessor, which zips and uploads the files to S3. The reason for appending the locale tag is that we have a record of each file uploaded for a locale, rather than each locale individually overwriting it's immediate predecessor.

Regarding the sync with Smartling it's only the file content that is used and the passed in fileName is used (not the file name set here on the object)